File: ehe/ehe.Web/Modules/DosyaUrunler/Urunler/UrunlerColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace ehe.DosyaUrunler.Columns;

[ColumnsScript("DosyaUrunler.Urunler")]
[BasedOnRow(typeof(UrunlerRow), CheckNames = true)]
public class UrunlerColumns
{
    //[EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    //public int UrunlerId { get; set; }
    [EditLink]
    public string UrunAdi { get; set; }
    [EditLink]
    public string UrunAciklamasi { get; set; }
    [EditLink]
    public int UrunStokAdeti { get; set; }
    [EditLink]
    public int UrunFiyati { get; set; }
    [EditLink]
    public int UrunKodu { get; set; }

    [QuickFilter, EditLink]
    public string UrunCesitleriidAd { get; set; }
   

}

File: ehe/ehe.Web/Modules/ServerTypes/DosyaUrunler/UrunlerColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { UrunlerRow } from "./UrunlerRow";

export interface UrunlerColumns {
    UrunAdi: Column<UrunlerRow>;
    UrunAciklamasi: Column<UrunlerRow>;
    UrunStokAdeti: Column<UrunlerRow>;
    UrunFiyati: Column<UrunlerRow>;
    UrunKodu: Column<UrunlerRow>;
    UrunCesitleriidAd: Column<UrunlerRow>;
}

export class UrunlerColumns extends ColumnsBase<UrunlerRow> {
    static readonly columnsKey = 'DosyaUrunler.Urunler';
    static readonly Fields = fieldsProxy<UrunlerColumns>();
}